internal/service/organization/resource_organization_test.go

```diff
@@ -56,39 +63,71 @@ func testAccCheckMongoDBAtlasOrganizationExists(resourceName string) resource.Te
 			return fmt.Errorf("no ID is set")
 		}
 
+		if rs.Primary.Attributes["public_key"] == "" {
+			return fmt.Errorf("no public_key is set")
+		}
+
+		if rs.Primary.Attributes["private_key"] == "" {
+			return fmt.Errorf("no private_key is set")
+		}
+
 		ids := conversion.DecodeStateID(rs.Primary.ID)
 
-		organizationOptions := &matlas.OrganizationsListOptions{}
-		orgs, _, err := conn.Organizations.List(context.Background(), organizationOptions)
+		cfg := config.Config{
+			PublicKey:  rs.Primary.Attributes["public_key"],
+			PrivateKey: rs.Primary.Attributes["private_key"],
+			BaseURL:    acc.TestAccProviderSdkV2.Meta().(*config.MongoDBClient).Config.BaseURL,
+		}
+
+		ctx := context.Background()
+		clients, _ := cfg.NewClient(ctx)
+		conn := clients.(*config.MongoDBClient).AtlasV2
+
+		orgs, _, err := conn.OrganizationsApi.ListOrganizations(ctx).Execute()
 		if err == nil {
-			for _, val := range orgs.Results {
-				if val.ID == ids["org_id"] {
-					return fmt.Errorf("Organization (%s) still exists", ids["role_name"])
+			for _, val := range *orgs.Results {
+				if *val.Id == ids["org_id"] {
+					return nil
 				}
 			}
-			return nil
+			return fmt.Errorf("Organization (%s) doesn't exist", ids["org_id"])
```

Comment:
can you clarify the change in the test context here? are we changing the nature of the error from "an org still exist" to "an org doesn't exist"? And why so?

Reply:
this method is supposed to check if the organization exists (return error otherwise), instead it was returning an error if the organization existed. 
Since the API keys being used to call the ListOrganizations() was not associated with the newly created org, it would not be returned, hence we now use the API keys of the new org - https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/2023-11-15/#tag/Organizations/operation/listOrganizations `To use this resource, the requesting API Key must have the Organization Member role`
Will add a comment